src/Components/LoginComponent/SocialLogin.js

```diff
@@ -0,0 +1,307 @@
+import Apiresource from "../../Config";
+
+function SocialLogin(props, visible) { 
+    const [ id, setId ] = useState('');
+    const [ name, setName ] = useState('');
+    const [ provider, setProvider ] = useState('');
+
+// 구글 로그인
+    const clickGoogleBtn = (res) => {
+        fetch(`${Apiresource}/account/sign-in/google`, {
+            method: 'POST',
+            headers: {
+                'content-type': 'application/json',
+              },
+            body: JSON.stringify({
+                email: res.profileObj.email,
+                name: res.profileObj.name,
+            })
+        })
+        .then(res => res.json())
+        .then(res => {
+            if (res) {
+                localStorage.setItem('kakao-token', res.token)
+                alert('로그인을 환영합니다')
+                props.history.push('/main')
+            } else {
+                alert('아이디와 비밀번호를 확인해주세요.')
+            }
+        })
+    }       
+
+// 카카오 로그인
+    const clickKakaoBtn = (res) => {
+        fetch(`${Apiresource}/account/sign-in/kakao`, {
+            method: 'POST',
+            headers: {
+                'content-type': 'application/json',
+              },
+            body: JSON.stringify({
+                access_token : res.response.access_token
+            })
+        })
+        .then(res => res.json())
+        .then(res => {
+            if (res.access_token) {
+            localStorage.setItem('kakao-token', res.access_token)
+            alert('로그인을 환영합니다')
+            props.history.push('/main')
+        } else {
+            alert('아이디와 비밀번호를 확인해주세요.')
+        }
+        })
+    }
+
+    const responseFail = (err) => {
+        console.error(err)
+    };
+
+    const clickBackBtn = () => {
+        props.history.push('/EmailLogin')
+        };
+        
+    const returnMain = () => {
+        props.history.push('/Main')
+        };
+
+    return(
+        <>
+            <ModalOverlay visible={visible} />
+            <ModalWrapper tabIndex="-1" visible={visible}>
+                <ModalIner tabIndex="0">
+                    <div>
+                        <LoginImageContainer>
+                            <LoginImgtag src="https://static.tacdn.com/img2/brand_refresh/Tripadvisor_lockup_horizontal_registered.svg" />
+                        </LoginImageContainer>
+                        <LogInnput>
+                            <LoginInputMargin>
+                                <div className="centerMargin">
+                                    <GoogleBtn
+                                        icon={false}
+                                        clientId={'323741016826-mmcgfi7be1gj5a2ki3n81ucmqhr8iagf.apps.googleusercontent.com'}
+                                        onSuccess={clickGoogleBtn}
+                                        onFailure={responseFail}
+                                        >    
+                                        <SocialImg src="https://static.tacdn.com/img2/google/G_color_40x40.png" />
```

Comment:
```suggestion
                                        <SocialImg type="google" />
```
이렇게 적어주시고 아래에서 props에 따라 이미지가 바뀌도록 해주세요

Reply:
이거 말씀하신대로 바꾸면 제가 이미지나 속성이 컨트롤이 안되어서 그렇습니다.
해당 내용을 컴포넌트로 불러왔는데, 제가 먹이는 css를 무시하고 구글 자체의 css를 먹다보니
어쩔 수 없이 이렇게 진행하였습니다.